pkg/client/interface.go

```diff
@@ -297,3 +298,10 @@ type SharedQueryClient interface {
 	// the session that includes queryHeight opens.
 	GetProofWindowOpenHeight(ctx context.Context, queryHeight int64) (int64, error)
 }
+
+// BlockQueryClient defines an interface that enables the querying of
```

Comment:
LGTM. No issues. Just calling out something below but not looking for a response.


Looks like we're already diverging from [1] w.r.t Naming (e.g. `BlockClient` vs `BlockQueryClient`)

I know:
- It has different uses
- It has different contexts
- Functionality differs

Just calling it out.

cc @adshmh @red-0ne 

[1] https://github.com/pokt-network/shannon-sdk/blob/d6cb20a34a2e50ea26f196f7c671e60a66bbc015/block.go

Reply:
Just note that `shannon-sdk`'s block client and `poktroll`'s block client will diverge in functionality.

While the former will mainly be used to query block heights, the later wants the full block to get block events and txs.

So they are definitely not interchangeable.